lattice_controller_engine/lib/lattice_controller_engine/observed/lattice.ex

```diff
@@ -0,0 +1,474 @@
+defmodule LatticeControllerEngine.Observed.Lattice do
+  @annotation_app_spec "wasmcloud.dev/appspec"
+
+  @moduledoc """
+  The root structure of an observed lattice. An observed lattice is essentially an
+  event sourced aggregate who state is determined by application of a stream of
+  lattice events
+  """
+  alias __MODULE__
+  alias LatticeControllerEngine.Observed.{Provider, Host, Instance, LinkDefinition}
+
+  # We need the keys to be there, even if they hold empty lists
+  @enforce_keys [:actors, :providers, :hosts, :linkdefs]
+  defstruct [:actors, :providers, :hosts, :linkdefs, :ocimap, :instance_tracking]
```

Comment:
Do we need at least empty maps for the ocimap and instance tracking too?

Reply:
Yes, we get those when we use the `Lattice.new` function